mods/cnc/maps/gdi09/gdi09-AI.lua

```diff
@@ -0,0 +1,201 @@
+	return #harv > 0
+end
+
+GuardBase = function()
+	Utils.Do(NodBase, function(building)
+		Trigger.OnDamaged(building, function()
+			Utils.Do(NodGuards, function(guard)
+				if not guard.IsDead and not building.IsDead then
+					guard.Stop()
+					guard.Guard(building)
+				end
+			end)
+		end)
+	end)
+end
+
+ProduceHarvester = function(building)
+	if not buildingHarvester then
+		buildingHarvester = true
+		building.Build(HarvesterProductionType, function()
+			buildingHarvester = false
+		end)
+	end
+end
+
+ProduceInfantry = function(building)
+	if building.IsDead or building.Owner ~= Nod then
+		return
+	elseif not CheckForHarvester() then
+		Trigger.AfterDelay(DateTime.Seconds(10), function() ProduceInfantry(building) end)
+		return
+	end
+
+	if #PatrolProductionQueue >= 1 then
+		local inQueue = PatrolProductionQueue[1]
+		local toBuild =  { inQueue.unit[1] }  
+		local patrolPath = inQueue.waypoints
+		building.Build(toBuild, function(unit)
+			ReplenishPatrolUnit(unit[1], handofnod, patrolPath, 40)
+			table.remove(PatrolProductionQueue, 1)
+		end)
+		Trigger.AfterDelay(DateTime.Seconds(10), function() ProduceInfantry(building) end)
+		return
+	end
+
+	local delay = Utils.RandomInteger(DateTime.Seconds(3), DateTime.Seconds(9))
+	local toBuild = { Utils.Random(InfantryProductionTypes) }
+	local Path = Utils.Random(AttackPaths)
+	building.Build(toBuild, function(unit)
+		InfantryAttackGroup[#InfantryAttackGroup + 1] = unit[1]
+
+		if #InfantryAttackGroup >= InfantryGroupSize then
+			MoveAndHunt(InfantryAttackGroup, Path)
+			InfantryAttackGroup = { }
+			Trigger.AfterDelay(InfantryProductionCooldown, function() ProduceInfantry(building) end)
+		else
+			Trigger.AfterDelay(delay, function() ProduceInfantry(building) end)
+		end
+	end)
+
+end
+
+ProduceVehicle = function(building)
+	if building.IsDead or building.Owner ~= Nod then
+		return
+	elseif not CheckForHarvester() then
+		ProduceHarvester(building)
+		Trigger.AfterDelay(DateTime.Seconds(10), function() ProduceVehicle(building) end)
+		return
+	end
+
+	local delay = Utils.RandomInteger(DateTime.Seconds(12), DateTime.Seconds(17))
+	local toBuild = { Utils.Random(VehicleProductionTypes) }
+	local Path = Utils.Random(AttackPaths)
+	building.Build(toBuild, function(unit)
+		VehicleAttackGroup[#VehicleAttackGroup + 1] = unit[1]
+
+		if #VehicleAttackGroup >= VehicleGroupSize then
+			MoveAndHunt(VehicleAttackGroup, Path)
+			VehicleAttackGroup = { }
+			Trigger.AfterDelay(VehicleProductionCooldown, function() ProduceVehicle(building) end)
+		else
+			Trigger.AfterDelay(delay, function() ProduceVehicle(building) end)
+		end
+	end)
+end
+
+StartAI = function()
+	RepairNamedActors(Nod, 0.75)
+
+	Nod.Cash = StartingCash
+	GuardBase()
+end
+
+Trigger.OnAllKilledOrCaptured(NodBase, function()
+	Utils.Do(GDI.GetGroundAttackers(), IdleHunt)
+end)
+
+rigger.OnKilled(nodrefinery, function(building)
+	BuildBuilding(BaseRefinery, NodCYard)
+end)
+
+Trigger.OnKilled(nodpower1, function(building)
```

Comment:
I guess we can also scratch that building variable in all scripts. But that would be a different PR
Trigger.OnKilled(nodpower1, function(~~building~~)